main.py

```diff
@@ -0,0 +1,179 @@
+        self.square_black = pygame.image.load("images/black.jpg")
+        self.square_red = pygame.image.load("images/red.png")
+        self.white_horse = pygame.image.load("images/white_horse.png")
+        self.black_horse = pygame.image.load("images/black_horse.png")
+        self.cursor = Cursor()
+        self.board = chessboard()
+        self.node = Node()
+        self.run()
+
+    def run(self):
+        """
+        Autor: Carlos Almario
+        Fecha: Mayo 16 2018
+        Método para correr la ventana del juego y escuchar los eventos que suceden en ella.
+        """
+
+        c = 0
+
+        start = False
+        while True:
+
+            pygame.display.update()
+            self.cursor.update()
+            # condición para pintar el tablero
+            if start:
+                self.window.fill(self.background_window)
+                self.rewrite()
+
+                possible_movimenents = self.node.possible_movements(self.board)
+
+                for i in range(len(self.board)):
+                    for j in range(len(self.board)):
+
+                        if (i+j) % 2 != 0:
+                            self.window.blit(self.square_black, (j*80,i*80))
+                        else:
+                            self.window.blit(self.square_white, (j * 80, i * 80))
+
+                        if self.board[i][j] == 'W':
+                            self.window.blit(self.white_horse, (j * 86, i * 80))
+
+                        if self.board[i][j] == 'B':
+                            self.window.blit(self.black_horse, (j * 86, i * 80))
+
+                for iterator in range(len(possible_movimenents)):
+                    self.window.blit(self.square_red,
+                                 (possible_movimenents[iterator][1] * 80, possible_movimenents[iterator][0] * 80))
+            else:
+                self.window.fill(self.background_two_window)
+                button = Image((WIDTH-297)/2, (HIGH-192)/2, "images/button.png")
+                button.draw(self.window)
+
+            # cliclo For que está escuchando los eventos
+            for event in pygame.event.get():
+
+                if event.type == pygame.MOUSEBUTTONDOWN:
+
+                    # condicion para el inicio del juego
+                    if start:
+                        position_x, position_y = int(self.cursor.top/80), int(self.cursor.left/80)
+
+                        # condicion que evalua si donde se da click es una casila del tablero
+                        if position_x in range(6) and position_y in range(6):
+                            """
+                            bucle que busca en la lista de posibles movimientos si la casilla donde se da cick es un
+                            movimiento posible
+                            """
+                            for iterator in range(len(possible_movimenents)):
+                                position_list_x = possible_movimenents[iterator][0]
```

Comment:
que es iterator?

Reply:
iterator en español es iterador en la variable que va cambiando en el ciclo for, es decir el indice de que itera para acceder a as posiciones de la lista